manifest/provider/waiter.go

```diff
@@ -300,3 +313,67 @@ func (w *RolloutWaiter) Wait(ctx context.Context) error {
 	w.logger.Info("[ApplyResourceChange][Wait] Rollout complete\n")
 	return nil
 }
+
+// ConditionsWaiter will wait for the specified conditions on
+// the resource to be met
+type ConditionsWaiter struct {
+	resource     dynamic.ResourceInterface
+	resourceName string
+	conditions   []tftypes.Value
+	logger       hclog.Logger
+}
+
+// Wait checks all the configured conditions have been met
+func (w *ConditionsWaiter) Wait(ctx context.Context) error {
+	w.logger.Info("[ApplyResourceChange][Wait] Waiting for conditions...\n")
+
+	for {
+		if deadline, ok := ctx.Deadline(); ok {
+			if time.Now().After(deadline) {
+				return context.DeadlineExceeded
+			}
+		}
+
+		res, err := w.resource.Get(ctx, w.resourceName, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if errors.IsGone(err) {
+			return fmt.Errorf("resource was deleted")
+		}
+
+		status := res.Object["status"].(map[string]interface{})
+		conditions := status["conditions"].([]interface{})
+		conditionsMet := map[string]bool{}
+		for _, c := range w.conditions {
+			var condition map[string]tftypes.Value
+			c.As(&condition)
+			var conditionType, conditionStatus string
+			condition["type"].As(&conditionType)
+			condition["status"].As(&conditionStatus)
+			conditionsMet[conditionType] = false
+			for _, cc := range conditions {
+				ccc := cc.(map[string]interface{})
+				if ccc["type"].(string) != conditionType {
+					continue
+				}
+				conditionsMet[conditionType] = ccc["status"].(string) == conditionStatus
+			}
+		}
+
+		done := true
+		for _, v := range conditionsMet {
+			if !v {
+				done = false
+			}
+		}
```

Comment:
This wouldn't be needed when using accumulator boolean. Instead of "done", the check would be on directly "conditionsMet".